internal/importers/base/importer.go

```diff
@@ -291,3 +299,20 @@ func (i *Importer) convertToConstraint(v gps.Version) gps.Constraint {
 	}
 	return v
 }
+
+func (i *Importer) isDefaultSource(projectRoot gps.ProjectRoot, sourceURL string) (bool, error) {
+	if sourceURL == "https://"+string(projectRoot) {
+		return true, nil
+	}
+
+	sourceURLs, err := i.sm.SourceURLsForPath(string(projectRoot))
+	if err != nil {
+		return false, err
+	}
+	// The first url in the slice will be the default one (usually https://...)
+	if len(sourceURLs) > 0 && sourceURL == sourceURLs[0].String() {
```

Comment:
If the first entry in `sourceURLs` doesn't exist, does dep try the other ones? e.g. If the imported source matched the second entry, but we don't import it and left the source blank, would dep automatically try the first, realize it doesn't exist and fall back to that second entry?

I am a bit confused as to why only the first entry is the only "default". I thought they all were defaults that dep would try?

Reply:
regarding you first question, yes, I believe dep will try the next URL in order, if one doesn't exist.

I compare only to the first one following the guidance of @sdboyer - good change I misunderstood something though. My guess for this, is that if a tool choose not the `https:` URL, but the `ssh` one for example, we better respect that and not try fetching via `https` even if it works.

Reply:
Sounds good to me, thanks!